Список контактов и детали контакта. Верстка и навигация

iaroslav18 opened the pull request:
Добавлен фрагмент со списком контактов.
В списке реализована одна карточка контакта.
Добавлен фрагмент с деталями контакта.
Реализован переход между фрагментами.

hex цветов:
#4a1824 - colorPrimary
#735552 - colorAccent
![contactList](https://user-images.githubusercontent.com/58564569/80705326-e8c07d00-8aee-11ea-9246-5afd3076c59b.jpg)
![contactDetails](https://user-images.githubusercontent.com/58564569/80621524-173b4b00-8a50-11ea-8a16-bb5e62c5692e.jpg)

iaroslav18 committed:
Добавлен фрагмент со списком контактов.
В списке реализована одна карточка контакта.
Добавлен фрагмент с деталями контакта.
Реализован переход между фрагментами.

goblinr commented on app/src/main/java/com/gmail/yarolave/lesson/ContactDetailsFragment.java:
```diff
@@ -0,0 +1,65 @@
+     * @param param2 Parameter 2.
+     * @return A new instance of fragment ContactDetailsFragment.
+     */
+    // TODO: Rename and change types and number of parameters
```
Это каким-то генератором было создано?

iaroslav18 replied:
Это было создано AndroidStudio

goblinr commented on app/src/main/java/com/gmail/yarolave/lesson/ContactDetailsFragment.java:
```diff
@@ -0,0 +1,65 @@
+        super.onCreate(savedInstanceState);
+        if (getArguments() != null) {
+            mParam1 = getArguments().getString(ARG_PARAM1);
+            mParam2 = getArguments().getString(ARG_PARAM2);
```
Какой смысл в этих 2-х параметрах? 
В требованиях к ДЗ нужно было передать только 1 параметр - ID контакта

goblinr commented on app/src/main/java/com/gmail/yarolave/lesson/ContactListFragment.java:
```diff
@@ -0,0 +1,79 @@
+     * @return A new instance of fragment ContactDetailsFragment.
+     */
+    // TODO: Rename and change types and number of parameters
+    public static ContactDetailsFragment newInstance(String param1, String param2) {
```
Здесь явно копипаста.
Для ContactListFragment не нужны параметры

goblinr commented on app/src/main/java/com/gmail/yarolave/lesson/ContactListFragment.java:
```diff
@@ -0,0 +1,79 @@
+
+    @Override
+    public void onClick(View v) {
+        ContactDetailsFragment details = new ContactDetailsFragment();
```
Нужно передавать ID выбранного контакта в аргументы ContactDetailsFragment

goblinr commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -6,13 +6,13 @@
-        android:layout_width="wrap_content"
-        android:layout_height="wrap_content"
-        android:text="Hello World!"
+    <fragment
```
Тэг `<fragment>` нельзя использовать совместно с динамическим добавлением фрагментов

goblinr commented on app/src/main/res/layout/fragment_contact_details.xml:
```diff
@@ -0,0 +1,111 @@
+        <TextView
+            android:id="@+id/firstNumber"
+            android:layout_width="match_parent"
+            android:layout_height="35dp"
```
Нечетных размеров следует избегать.
Это касается всего проекта

goblinr commented on app/src/main/res/layout/fragment_contact_list.xml:
```diff
@@ -0,0 +1,57 @@
+    android:layout_height="match_parent"
+    tools:context=".ContactListFragment">
+
+    <LinearLayout
```
Бессмысленное вложение других контейнеров в `ConstraintLayout`
Если используешь `ConstraintLayout` - то делай плоскую структуру - никаких вложений 3-го и более уровня

iaroslav18 committed:
Исправлены замечания из первого PR

goblinr commented on app/src/main/java/com/gmail/yarolave/lesson/ContactDetailsFragment.java:
```diff
@@ -0,0 +1,38 @@
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
```
Параметр paramId  нужно восстанавливать из аргументов

goblinr commented on app/src/main/res/layout/fragment_contact_list.xml:
```diff
@@ -0,0 +1,53 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Бессмысленный LinearLayout всего с одним элементом внутри

goblinr replied:
Теперь бессмысленный ConstraintLayout. Если используешь ConstraintLayout, то структуру внутри него нужно делать плоской, т.е. не должно быть вложенности больше одной.

iaroslav18 committed:
Исправлены замечания:
Бесполезный LinearLayout в ContactListFragment
paramId должен восстанавливаться из аргументов

iaroslav18 committed:
Убрана излишняя вложенность

goblinr reviewed (approved):
Обнови скриншоты и можно мержить
